src/components/layouts/ToolLayout/index.tsx

```diff
@@ -1,46 +1,65 @@
-import React from "react";
-import { Box, ChakraProvider, Flex, IconButton, Image } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import {
+    Box,
+    Button,
+    ChakraProvider,
+    Flex,
+    IconButton,
+    Image,
+} from "@chakra-ui/react";
 import { MdMenu } from "react-icons/md";
+import { useRouter } from "next/router";
 
 import { defaultTheme } from "@definitions/chakra/theme";
 import { getIsDesktop } from "src/utils/media/mediaHelpers";
+import { Auth } from "aws-amplify";
 
 export const ToolLayout: React.FC = ({ children }) => {
     const [isDesktop] = getIsDesktop();
+    const [user, setUser] = useState<any | null>(null);
+
+    useEffect(() => {
+        const getCurrentUser = async () => {
+            const currentUser = await Auth.currentAuthenticatedUser();
+            setUser(currentUser);
+        };
+
+        getCurrentUser().catch(console.error);
+    }, []);
+    const router = useRouter();
+
+    const Logo = () => (
+        <Image
+            onClick={() => router.push("/")}
+            src="/assets/hug-logo.png"
+            cursor={"pointer"}
+        ></Image>
```

Comment:
nit: maybe this could be
```suggestion
        <Image
            onClick={() => router.push("/")}
            src="/assets/hug-logo.png"
            cursor={"pointer"}
        />
```